app/lib/utils/gpDataMapper.js

```diff
@@ -3,4 +3,58 @@ function getBookOnlineLink(gpData) {
     ? gpData.onlineServices.appointments.url : undefined;
 }
 
-module.exports = getBookOnlineLink;
+function getFilteredGps(gpData, searchStr) {
+  if (gpData.doctors) {
+    let filteredSearchStr = searchStr;
+    let filteredDocs = [];
+    if (searchStr.match(new RegExp(/^(doctor|dr.|dr)/, 'i'))) {
```

Comment:
Use the literal form of regex creation as opposed to constructor form when the input isn't runtime specific